### ISSUE TITLE
Run time warning - invalid value encountered in power

### ISSUE THREAD

**Issue by svadams** [SpiNNakerManchester/sPyNNaker#99]
I get a warning apparently generated from key_field_generator.py line 20 which says

RuntimeWarning: invalid value encountered in power:

self._n_mask_keys = 2 ** zeros

Actually this does not seem to be causing a problem as my simulation runs and does the right
thing but I thought I'd report it to find out if I should be concerned!

**Comment by neworderofjamie** [comment]
I have this too!

**Comment by dr-david-lester** [comment]
Could it be an issue with x ** y being evaluated with one or both of x and y being 0.0?

Because 0.0 ** 0.0 is either 1.0 or 0.0 (depending on how you look at it), but 0 ** 0 is
always 1.

I’m a bit mystified by 2.0 ** 0.0 causing problems. That ought to be 1.0 under all conceivable interpretations!

TL; DR change zeros to be integer, and/or call integer operator for (**).

Dave

On 1 Jul 2015, at 15:56, neworderofjamie <notifications@github.com<mailto:notifications@github.com>> wrote:

I have this too!

**Comment by rowleya** [comment]
I can’t remember having seen this but I have found where it is in the Python code.  The line should read:
self._n_key_masks = 2 ** len(zeros)

zeros is an array of the positions of the zeros in the mask, hence the error.  How anything still works after this is anyone’s guess!  I will fix it...

Andrew :)

**Comment by rowleya** [comment]
Fixed in master
